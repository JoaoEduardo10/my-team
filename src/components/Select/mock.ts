/* eslint-disable @typescript-eslint/no-unused-vars */
import { TCountrys, TLeagues, TTeam } from '../../types';

const mockSeason: number[] = [2022, 2010, 2020];

const mockContrys:TCountrys['response'] = [
  {
    code: 'T1',
    flag: 'test',
    name:  'test 1'
  },
  {
    code: 'T2',
    flag: 'test 2',
    name:  'test 2'
  },
  {
    code: 'T3',
    flag: 'test3',
    name:  'test 3'
  },
];


const mockLeague: TLeagues['response'] = [
  {
    country: {
      code: 'T1',
      flag: 'test',
      name: 'Test 1'
    },
    league: {
      id: 1,
      logo: 'tes.png',
      name: 'test ligue',
      type: 'test'
    },
    seasons: [
      {
        current: 'test',
        end: '10/10/2022',
        start: '01/01/2022',
        year: 2022
      }
    ]
  }
];

const mockTeam: TTeam['response'] = [
  {
    team: {
      code: 'br',
      country: 'brazil',
      id: 1,
      logo: 'test.png',
      name: 'bar√ßa'
    }
  }
];

export { mockContrys, mockLeague, mockSeason, mockTeam };
