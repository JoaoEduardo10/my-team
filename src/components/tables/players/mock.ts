export const mockPlayers = [
  {
    player: {
      age: 18,
      id: 1,
      name: 'eduardo',
      nationality: 'brasileiro'
    }
  },
  {
    player: {
      age: 20,
      id: 2,
      name: 'felipe',
      nationality: 'brasileiro'
    }
  },
  {
    player: {
      age: 20,
      id: 3,
      name: 'caio',
      nationality: 'brasileiro'
    }
  },
  {
    player: {
      age: 30,
      id: 4,
      name: 'renato',
      nationality: 'brasileiro'
    }
  },
];
