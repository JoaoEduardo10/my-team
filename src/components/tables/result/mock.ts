import { TTeamStatistic } from '../../../types';

export const mockResult: TTeamStatistic['response'] = {
  fixtures: {
    draws: {
      away: 4,
      home: 4,
      total: 8
    },
    loses:{
      away: 1,
      home: 4,
      total: 5
    },
    played: {
      away: 1,
      home: 10,
      total: 29
    },
    wins:{
      away: 1,
      home: 5,
      total: 6
    }
  },
  goals: {
    against: {
      minute: {
        '0-15': { percentage: '32.5', total: 2 },
        '106-120': { percentage: '32.5', total: 2 },
        '16-30': { percentage: '32.5', total: 2 },
        '31-45': { percentage: '32.5', total: 2 },
        '46-60': { percentage: '32.5', total: 2 },
        '61-75': { percentage: '32.5', total: 2 },
        '76-90': { percentage: '32.5', total: 2 },
        '91-105': { percentage: '32.5', total: 2 },
      }
    }
  },
  lineups: [
    {
      formation: '4-3-3',
      played: 7
    },
    {
      formation: '4-4-3',
      played: 10
    },
    {
      formation: '4-5-2',
      played: 13
    },
  ],
  team: {
    id: 12,
    logo: 'Saopaulo.png',
    name: 'SÃ£o Paulo'
  }
};
