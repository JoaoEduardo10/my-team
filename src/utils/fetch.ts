/* eslint-disable @typescript-eslint/no-explicit-any */

import { TCountrys, TLeagues, TTeam } from '../types';

const url = import.meta.env.VITE_URL_API as string;

const authUser = async (key: string) => {
  const response = await fetch(`${url}/status`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data = await response.json();

  return data;

};

const getCountry = async (key: string) => {
  const response = await fetch(`${url}/countries`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: TCountrys = await response.json();

  return data;
};

const getLeague= async (key: string, country: string, season: number) => {
  const response = await fetch(`${url}/leagues?country=${country}&season=${season}`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: TLeagues = await response.json();

  return data;
};

const getSeason = async (key: string) => {
  const response = await fetch(`${url}/leagues/seasons`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: { response: number[] } = await response.json();

  return data;
};

const getTeam = async (key: string, season: number, league: number) => {
  const response = await fetch(`${url}/teams?league=${league}&season=${season}`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: TTeam = await response.json();

  return data;
};


export { authUser, getCountry, getLeague, getSeason, getTeam };
