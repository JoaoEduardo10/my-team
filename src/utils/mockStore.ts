import configureStore from 'redux-mock-store';


const mockStore = configureStore([]);

const store = mockStore({
  team: {
    country: 'brasil',
    season: '2022',
    league: '47',
    team: 'SÃ£o paulo'
  }
});

export { store };
