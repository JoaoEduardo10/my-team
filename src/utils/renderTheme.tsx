import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import { GlobalStyles } from '../styles/globals-styles';
import { RouterProvider, createBrowserRouter } from 'react-router-dom';
import { AuthProvider } from '../context/auth';
import { Provider } from 'react-redux';
import { store } from './mockStore';

const mockRouet = (children: React.ReactNode, path?: string) => {
  return createBrowserRouter([
    {
      path: path ?? '/',
      element: children
    }
  ]);
};

export const renderTheme = (
  children: React.ReactNode,
  path?: string
) => {
  return render(
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <AuthProvider>
          <RouterProvider router={mockRouet(children, path)} />
        </AuthProvider>
        <GlobalStyles />
      </ThemeProvider>
    </Provider>
  );
};
