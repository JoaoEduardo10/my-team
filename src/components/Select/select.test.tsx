import { vi } from 'vitest';
import { Select } from '.';
import { renderTheme } from '../../utils/renderTheme';
import { mockContrys, mockLeague, mockSeason, mockTeam } from './mock';
import { fireEvent, screen } from '@testing-library/react';

describe('<Select />', () => {
  const onChange = vi.fn();

  it('should return the country select', () => {
    renderTheme(<Select loading={false} arrey={mockContrys} type='country' onChange={onChange} value='test' />);

    const allCountry = screen.getAllByLabelText('Lista De Paises');

    expect(allCountry.length).toEqual(3);
  });

  it('should return the League select', () => {
    renderTheme(<Select loading={false} arrey={mockLeague} type='league' onChange={onChange} value='test' />);

    const allCountry = screen.getAllByLabelText('Lista De Ligas');

    expect(allCountry.length).toEqual(1);
  });

  it('should return the Season select', () => {
    renderTheme(<Select loading={false} arrey={mockSeason} type='season' onChange={onChange} value='test' />);

    const allCountry = screen.getAllByLabelText('Lista De Temporadas');

    expect(allCountry.length).toEqual(3);
  });

  it('should return the Team select', () => {
    renderTheme(<Select loading={false} arrey={mockTeam} type='team' onChange={onChange} value='test' />);

    const allCountry = screen.getAllByLabelText('Lista De Times');
    const message = screen.getByRole('alert');

    expect(message).toHaveTextContent('Selecione um time');
    expect(allCountry.length).toEqual(1);
  });

  it('should show the team loading message', () => {
    renderTheme(<Select loading={true} arrey={mockTeam} type='team' onChange={onChange} value='test' />);

    const message = screen.getByRole('alert');
    expect(message).toHaveTextContent('carregando...');
  });

  it('should show the Season loading message', () => {
    renderTheme(<Select loading={true} arrey={mockSeason} type='season' onChange={onChange} value='test' />);

    const message = screen.getByRole('alert');
    expect(message).toHaveTextContent('carregando...');
  });

  it('should show the League loading message', () => {
    renderTheme(<Select loading={true} arrey={mockLeague} type='league' onChange={onChange} value='test' />);

    const message = screen.getByRole('alert');
    expect(message).toHaveTextContent('carregando...');
  });

  it('should show the country loading message', () => {
    renderTheme(<Select loading={true} arrey={mockContrys} type='country' onChange={onChange} value='test' />);

    const message = screen.getByRole('alert');
    expect(message).toHaveTextContent('carregando...');
  });

  it('should get the select value', () => {
    renderTheme(<Select loading={false} arrey={mockTeam} type='team' onChange={onChange} value='test' />);

    const allCountry = screen.getAllByLabelText('Lista De Times');

    expect(allCountry.length).toEqual(1);

    const select = screen.getByRole('listbox');

    fireEvent.change(select);

  });

  it('should to match snapshort', () => {
    const {container } =  renderTheme(<Select loading={false} arrey={mockTeam} type='team' onChange={onChange} value='test' />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
