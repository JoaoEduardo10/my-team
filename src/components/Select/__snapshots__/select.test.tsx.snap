// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Select /> > should to match snapshort 1`] = `
<select
  class="fcDRZf"
  role="listbox"
>
  <option
    class="cvjFPq"
    disabled=""
    role="alert"
    value=""
  >
    Selecione um time
  </option>
  <option
    aria-label="Lista De Times"
    class="cvjFPq"
    value="1"
  >
    bar√ßa
  </option>
</select>
`;
